emission/tests/exportTests/TestExportPipelineConfirmedTripsandBackground.py

```diff
@@ -0,0 +1,75 @@
+from future import standard_library
+standard_library.install_aliases()
+from builtins import *
+import unittest
+import json
+import bson.json_util as bju
+import emission.storage.timeseries.abstract_timeseries as esta
+import gzip
+import glob
+import emission.tests.common as etc
+import emission.pipeline.export_stage as epe
+import emission.storage.pipeline_queries as espq
+import emission.exportdata.export_data as eeed
+
+class TestExportPipelineConfirmedTripsandBackground(unittest.TestCase):
+    def testExportPipeline(self):
+        etc.setupRealExample(self, "emission/tests/data/real_examples/shankari_2015-07-22")
+        etc.runIntakePipeline(self.testUUID)
+        eeed.export_data(self.testUUID)
+        
+        #Data has been run through both pipelines, file has been exported.
+        tq = espq.get_time_range_for_export_data(self.testUUID)
+        exported_file_name = "emission/archived/archive_%s_%s_%s.gz" % (self.testUUID, tq.startTs, tq.endTs)
+        
+        #trim this file name from the back 
+        #generate a list of filesnames that are close (exports that are recent with this user, small list)
+	#this is a HACK, is there a better way? I am lost!(hoping list is 1)
+        
+        exported_file_name_trimmed = exported_file_name[:len(exported_file_name) - 15]
+        file_list = []
+        for name in glob.glob("%s*.gz" % exported_file_name_trimmed):
+            file_list.append(name)
+        for exported_file_name in file_list:
+            with gzip.open(exported_file_name, 'r') as ef:
+                exported_data = json.loads(ef.read().decode('utf-8'))
+            
```

Comment:
you are writing a unit test. for this part, you can just call `eee.export(` with a filename. Then you don't need to guess.

To test the pipeline step, you do need to figure out the expected filename and see if it exists. Not sure why this is a hack, this is the filename that you expect, no? If I were writing this, I would create a 

```
@staticmethod
def getFileName(user_id, time_query):
    return "emission/archived/archive_%s_%s_%s" % (user_id, time_query.startTs, time_query.endTs)
```

and then:
- test that separately
- use that to test the pipeline step
but I will not insist on that

Reply:
> Then you don't need to guess.
What do you mean by this? 

>Not sure why this is a hack, this is the filename that you expect, no?
The issue here seems to be getting a time_query that exactly matches the one used within the export pipeline. They seem to be off by a few digits within the time_query.end which is why I trimmed the output and searched for names that would match closely. I was having a hard time generating a filename that would exactly match.

Reply:
> Then you don't need to guess.
> What do you mean by this?

If you pass in a filename, you don't have to guess the filename. For unit testing, you don't need to call the top level functions (e.g. the intake stage), you can call internal functions (e.g. `eee.export(` as I suggested)

Reply:
Got it. So with the unit testing, I do not need to start off by running the raw data through the pipeline and then through the entire export pipeline. I can simply call `export(user_id, ts, start_ts, end_ts, file_name, ma_bool)` choosing the filename so I have it for testing. 
I am assuming then that the following functionality that exists within the `run_export_data_pipeline`:
`ts = esta.TimeSeries.get_time_series(user_id)
        time_query = espq.get_time_range_for_export_data(user_id)
        file_name = "emission/archived/archive_%s_%s_%s" % (user_id, time_query.startTs, time_query.endTs)
        eee.export(user_id, ts, time_query.startTs, time_query.endTs, file_name, False)
`
This way I can access the file_name within the testing infrastructure as well as the time_query. This will fix some of the matching issues I had as well. Thank you!